ding/entry/serial_entry_dqfd.py

```diff
@@ -0,0 +1,201 @@
+        expert_data[i]['is_expert'] = 1
+    expert_buffer.push(expert_data, cur_collector_envstep=0)
+    for _ in range(cfg.policy.learn.per_train_iter_k):
+        if evaluator.should_eval(learner.train_iter):
+            stop, reward = evaluator.eval(learner.save_checkpoint, learner.train_iter, collector.envstep)
+            if stop:
+                break
+        # Learn policy from collected data
+        # Expert_learner will train ``update_per_collect == 1`` times in one iteration.
+        train_data = expert_buffer.sample(learner.policy.get_attribute('batch_size'), learner.train_iter)
+        if train_data is None:
+            # It is possible that replay buffer's data count is too few to train ``update_per_collect`` times
+            logging.warning(
+                "Replay buffer's data can only train for {} steps. ".format(i) +
+                "You can modify data collect config, e.g. increasing n_sample, n_episode."
+            )
+            break
+        learner.train(train_data, collector.envstep)
+        if learner.policy.get_attribute('priority'):
+            expert_buffer.update(learner.priority_info)
+    learner.priority_info = {}
+    # Accumulate plenty of data at the beginning of training.
+    if cfg.policy.get('random_collect_size', 0) > 0:
+        action_space = collector_env.env_info().act_space
+        random_policy = PolicyFactory.get_random_policy(policy.collect_mode, action_space=action_space)
+        collector.reset_policy(random_policy)
+        collect_kwargs = commander.step()
+        new_data = collector.collect(n_sample=cfg.policy.random_collect_size, policy_kwargs=collect_kwargs)
+        for i in range(len(new_data)):
+            new_data[i]['is_expert'] = 0
+        replay_buffer.push(new_data, cur_collector_envstep=0)
+        collector.reset_policy(policy.collect_mode)
+    for _ in range(max_iterations):
+        collect_kwargs = commander.step()
+        # Evaluate policy performance
+        if evaluator.should_eval(learner.train_iter):
+            stop, reward = evaluator.eval(learner.save_checkpoint, learner.train_iter, collector.envstep)
+            if stop:
+                break
+        # Collect data by default config n_sample/n_episode
+        new_data = collector.collect(train_iter=learner.train_iter, policy_kwargs=collect_kwargs)
+        for i in range(len(new_data)):
+            new_data[i]['is_expert'] = 0
+        replay_buffer.push(new_data, cur_collector_envstep=collector.envstep)
+        # Learn policy from collected data
+        for i in range(cfg.policy.learn.update_per_collect):
+            # Learner will train ``update_per_collect`` times in one iteration.
+            # The hyperparameter pho, the demo ratio, control the propotion of data coming\
+            # from expert demonstrations versus from the agent's own experience.
+            stats = torch.rand((learner.policy.get_attribute('batch_size'))) < cfg.policy.collect.pho
```

Comment:
`np.random.choice` can be more user friendly

Reply:
Thx! corrected.